### ISSUE TITLE
Constant refresh of the page (salesforce.com + Do Not Track)

### ISSUE THREAD

**Issue by Acid-Crash** [sereneblue/chameleon#81]
Hi Here.
I have a strange issue with one of my internal web site (salesforce.com) that uses some sort of a timer.
https://success.salesforce.com/ideaview?id=087300000006txo
In FF63 Win10x64, Chameleon 0.95 (Profile > Real profile) the page keeps refreshing.
Once I tick on "Enable DNT (Do Not Track)" in extension settings the page starts working as expected.
Also In Firefox > Settings > Do Not Track is set to "Only when Firefox is set to block Detected Trackers".
If I switch to "Always" the page is also working correctly without any adjustments in extension settings

Tried to use a fresh Firefox profile (only Chameleon 0.95 is installed) and the issue is still present.
Unfortunately, due to security reasons, I can't share the full URL link, but maybe you still be able to investigate the issue.

**Comment by sereneblue** [comment]
Have you tried whitelisting salesforce (as a temporary solution)? I'm curious what's causing the issue.

**Comment by Acid-Crash** [comment]
Tried to make the following changes to the options.
No luck issue is still there.
The behavior stays the same, to be able to work with the page I need to enable "Do not Track" either in Firefox or in extension
![ua](https://user-images.githubusercontent.com/32600318/47412713-282d4100-d775-11e8-8a95-2caef5f794eb.png)

**Comment by sereneblue** [comment]
Is it that specific Salesforce app that's causing issues? Are there any errors shown in the console? I've never used Salesforce before so it's going to be difficult for me to replicate your issue.

**Comment by Acid-Crash** [comment]
I see.
Yes this app is present on a specific page.
Other Salesforce pages are working fine.
Here are few the console logs (fresh profile)

<details>

```
Navigated to https://*removed*removed*/5005000001AUjKH
Loading failed for the <script> with source “https://*removed*removed*/jslibrary/jslabels/1540418410000/en_US.js”. emptyHtmlDoc.html:1:1
Loading failed for the <script> with source “https://*removed*removed*/jslibrary/jslabels/1540418410000/en_US.js”. 5005000001AUjKH:11:1
SyntaxError: expected expression, got '<'[Learn More] jquery-1.7.1.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery-ui.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery.mousewheel.js:2
SyntaxError: expected expression, got '<'[Learn More] timer.js:2
ReferenceError: uniTabFrameLoad is not defined[Learn More] WorkItHome:1:1
onload
https://removed*removed.visual.force.com/apex/WorkItHome:1:1
Navigated to https://*removed*removed*/5005000001AUjKH
TypeError: b is undefined[Learn More] main.js:1002:150
DetailPage.prototype.handleRLAjaxResponse
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1002:150
DetailPage.prototype.makeRLAjaxRequest/<
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1001:436
b
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:365:54
Loading failed for the <script> with source “https://*removed*removed*/jslibrary/jslabels/1540418410000/en_US.js”. 5005000001AUjKH:11:1
SyntaxError: expected expression, got '<'[Learn More] jquery-1.7.1.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery-ui.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery.mousewheel.js:2
SyntaxError: expected expression, got '<'[Learn More] timer.js:2
ReferenceError: uniTabFrameLoad is not defined[Learn More] WorkItHome:1:1
onload
https://removed*removed.visual.force.com/apex/WorkItHome:1:1
Navigated to https://*removed*removed*/5005000001AUjKH
TypeError: b is undefined[Learn More] main.js:1002:150
DetailPage.prototype.handleRLAjaxResponse
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1002:150
DetailPage.prototype.makeRLAjaxRequest/<
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1001:436
b
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:365:54
Loading failed for the <script> with source “https://*removed*removed*/jslibrary/jslabels/1540418410000/en_US.js”. 5005000001AUjKH:11:1
SyntaxError: expected expression, got '<'[Learn More] jquery-1.7.1.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery-ui.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery.mousewheel.js:2
SyntaxError: expected expression, got '<'[Learn More] timer.js:2
ReferenceError: uniTabFrameLoad is not defined[Learn More] WorkItHome:1:1
onload
https://removed*removed.visual.force.com/apex/WorkItHome:1:1
Navigated to https://*removed*removed*/5005000001AUjKH
TypeError: b is undefined[Learn More] main.js:1002:150
DetailPage.prototype.handleRLAjaxResponse
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1002:150
DetailPage.prototype.makeRLAjaxRequest/<
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1001:436
b
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:365:54
Loading failed for the <script> with source “https://*removed*removed*/static/111213/js/perf/stub.js”.
```

</details>

<details>

```
Navigated to https://*removed*removed*/5005000001AUjKH
SyntaxError: expected expression, got '<'[Learn More] jquery-1.7.1.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery-ui.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery.mousewheel.js:2
SyntaxError: expected expression, got '<'[Learn More] timer.js:2
ReferenceError: uniTabFrameLoad is not defined[Learn More] WorkItHome:1:1
Navigated to https://*removed*removed*/5005000001AUjKH
TypeError: b is undefined[Learn More] main.js:1002:150
DetailPage.prototype.handleRLAjaxResponse
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1002:150
DetailPage.prototype.makeRLAjaxRequest/<
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1001:436
b
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:365:54
Loading failed for the <script> with source “https://*removed*removed*/jslibrary/jslabels/1540418410000/en_US.js”. 5005000001AUjKH:11:1
SyntaxError: expected expression, got '<'[Learn More] jquery-1.7.1.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery-ui.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery.mousewheel.js:2
SyntaxError: expected expression, got '<'[Learn More] timer.js:2
ReferenceError: uniTabFrameLoad is not defined[Learn More] WorkItHome:1:1
onload
https://removed*removed.na89.visual.force.com/apex/WorkItHome:1:1
Navigated to https://*removed*removed*/5005000001AUjKH
TypeError: b is undefined[Learn More] main.js:1002:150
DetailPage.prototype.handleRLAjaxResponse
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1002:150
DetailPage.prototype.makeRLAjaxRequest/<
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1001:436
b
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:365:54
Loading failed for the <script> with source “https://*removed*removed*/jslibrary/jslabels/1540418410000/en_US.js”. 5005000001AUjKH:11:1
SyntaxError: expected expression, got '<'[Learn More] jquery-1.7.1.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery-ui.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery.mousewheel.js:2
SyntaxError: expected expression, got '<'[Learn More] timer.js:2
ReferenceError: uniTabFrameLoad is not defined[Learn More] WorkItHome:1:1
onload
https://removed*removed.na89.visual.force.com/apex/WorkItHome:1:1
Navigated to https://*removed*removed*/5005000001AUjKH
TypeError: b is undefined[Learn More] main.js:1002:150
DetailPage.prototype.handleRLAjaxResponse
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1002:150
DetailPage.prototype.makeRLAjaxRequest/<
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:1001:436
b
https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js:365:54
Loading failed for the <script> with source “https://*removed*removed*/jslibrary/jslabels/1540418410000/en_US.js”. 5005000001AUjKH:11:1
SyntaxError: expected expression, got '<'[Learn More] jquery-1.7.1.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery-ui.min.js:2
SyntaxError: expected expression, got '<'[Learn More] jquery.mousewheel.js:2
SyntaxError: expected expression, got '<'[Learn More] timer.js:2
ReferenceError: uniTabFrameLoad is not defined[Learn More] WorkItHome:1:1
Source map error: SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data
Resource URL: https://*removed*removed*/jslibrary/1539646802000/sfdc/main.js
Source Map URL: /javascript/1540373783000/sfdc/source/main.js.map[Learn More]
```

</details>

**Comment by sereneblue** [comment]
Thanks for posting the output. Sadly, it's not too descriptive but I wonder if recent changes broke something.

Can you try testing some of the previous versions listed here https://addons.mozilla.org/en-US/firefox/addon/chameleon-ext/versions/ (specifically v0.9.2 or lower).

**Comment by sereneblue** [comment]
I believe v0.9.7 fixes this issue. Please test and let me know. :)

**Comment by Acid-Crash** [comment]
Hi,
I didn't have any issues with this site before 0.95 update.
Happy to confirm that the issue is indeed fixed in 0.97
Thank you for the help
Closing ticket
